src/analysis/normed_space/conformal.lean

```diff
@@ -0,0 +1,249 @@
+    rw [real_inner_smul_left, real_inner_smul_right, ← mul_assoc, inner_map_map],
+    repeat { rw [norm_smul, norm_map] },
+    rw [← mul_assoc],
+    exact calc c₁ * c₁ * ⟪u, v⟫ / (∥c₁∥ * ∥u∥ * ∥c₁∥ * ∥v∥)
+            = c₁ * c₁ * ⟪u, v⟫ / (∥c₁∥ * ∥c₁∥ * ∥u∥ * ∥v∥) : by simp only [mul_comm, mul_assoc]
+        ... = c₁ * c₁ * ⟪u, v⟫ / (abs c₁ * abs c₁ * ∥u∥ * ∥v∥) : by rw [real.norm_eq_abs]
+        ... = c₁ * c₁ * ⟪u, v⟫ / (c₁ * c₁ * ∥u∥ * ∥v∥) : by rw [← pow_two, ← sq_abs, pow_two]
+        ... = c₁ * (c₁ * ⟪u, v⟫) / (c₁ * (c₁ * (∥u∥ * ∥v∥))) : by simp only [mul_assoc]
+        ... = (c₁ * ⟪u, v⟫) / (c₁ * (∥u∥ * ∥v∥)) : by rw mul_div_mul_left _ _ hc₁
+        ... = ⟪u, v⟫ / (∥u∥ * ∥v∥) : by rw mul_div_mul_left _ _ hc₁, },
+end
+
+end is_conformal_map
+
+/-- A map `f` is said to be conformal if it has a conformal differential `f'`. -/
+def conformal_at {X Y : Type*}
+  [normed_group X] [normed_group Y] [normed_space ℝ X] [normed_space ℝ Y]
+  (f : X → Y) (x : X) :=
+∃ (f' : X →L[ℝ] Y), has_fderiv_at f f' x ∧ is_conformal_map ℝ f'
+
+lemma conformal_at_id {X : Type*}
+  [normed_group X] [normed_space ℝ X] (x : X) : conformal_at id x :=
+⟨id ℝ X, has_fderiv_at_id _, 1, one_ne_zero, id, by ext; simp⟩
+
+lemma conformal_at_const_smul {X : Type*}
+  [normed_group X] [normed_space ℝ X] {c : ℝ} (h : c ≠ 0) (x : X) :
+  conformal_at (λ (x': X), c • x') x :=
+⟨c • continuous_linear_map.id ℝ X, has_fderiv_at.const_smul (has_fderiv_at_id x) c,
+  c, h, id, by ext; simp⟩
+
+namespace conformal_at
+
+lemma differentiable_at {f : X → Y} {x : X} (h : conformal_at f x) :
+  differentiable_at ℝ f x :=
+let ⟨_, h₁, _, _, _, _⟩ := h in h₁.differentiable_at
+
+lemma comp {f : X → Y} {g : Y → Z} (x : X)
+  (hg : conformal_at g (f x)) (hf : conformal_at f x)  : conformal_at (g ∘ f) x :=
+begin
+  rcases hf with ⟨f', hf₁, cf, hcf, lif, hf₂⟩,
+  rcases hg with ⟨g', hg₁, cg, hcg, lig, hg₂⟩,
+  exact ⟨g'.comp f', has_fderiv_at.comp x hg₁ hf₁, cg * cf, mul_ne_zero hcg hcf, lig.comp lif,
+    by { ext, simp [coe_comp' f' g', hf₂, hg₂, smul_smul cg cf] }⟩,
+end
+
+lemma const_smul {f : X → Y} {x : X} {c : ℝ} (hc : c ≠ 0) (hf : conformal_at f x) :
+  conformal_at (c • f) x :=
+let p := (conformal_at_const_smul hc $ f x).comp x hf in
+begin
+  have : c • f = (λ x', c • x') ∘ f := by ext; simp only [pi.smul_apply, function.comp_app],
+  rw this, exact p,
+end
+
+/-- A real differentiable map `f` is conformal at point `x` if and only if
+    its differential `f'` at that point scales any inner product by a positive scalar. -/
+lemma _root_.conformal_at_iff {f : E → F} {x : E} {f' : E →L[ℝ] F}
+  (h : has_fderiv_at f f' x) : conformal_at f x ↔ ∃ (c : ℝ) (hc : 0 < c),
+  ∀ (u v : E), ⟪f' u, f' v⟫ = (c : ℝ) * ⟪u, v⟫ :=
+begin
+  split,
+  { rintros ⟨f₁, hf, c₁, hc₁, li, hf₁⟩,
+    exact (hf.unique h) ▸ (is_conformal_map_iff f₁).mp ⟨c₁, hc₁, li, hf₁⟩, },
+  { rintros ⟨c, hc, huv⟩,
+    exact ⟨f', h, (is_conformal_map_iff f').mpr ⟨c, hc, huv⟩⟩, },
```

Comment:
Just because  you can explode something into really small pieces doesn't mean you should - here you put things back together again exactly as they were, so it's easier to not break them apart at all:

```suggestion
  split,
  { rintros ⟨f₁, hf, hc⟩,
    exact (hf.unique h) ▸ (is_conformal_map_iff f₁).mp hc, },
  { rintros hc,
    exact ⟨f', h, (is_conformal_map_iff f').mpr hc⟩, },
```

Reply:
I would also try to prove this without a `split`.  This is an `iff` and `is_conformal_map_iff` is an `iff`, so it shouldn't be necessary.